File: 2015/Day_07/Python/test_part_1.py
from . import part_1


def test_create_dictionary():
    instructions = ['123 -> x', '456 -> y', 'x AND y -> d', 'x OR y -> e', 'x LSHIFT 2 -> f', 'y RSHIFT 2 -> g',
                    'NOT x -> h', 'NOT y -> i']
    wire_dictionary = part_1.create_dictionary(instructions)
    assert wire_dictionary['x'] == "123"
    assert wire_dictionary['y'] == "456"
    assert wire_dictionary['d'] == "x AND y"


def test_break_up_equation():
    equation = "x AND y"
    assert part_1.break_up_equation(equation) == ('x', 'AND', 'y')
    equation = "x LSHIFT 2"
    assert part_1.break_up_equation(equaretion) == ('x', "LSHIFT", '2')
    equation = "NOT x"
    assert part_1.break_up_equation(equation) == ("NOT", 'x')


def test_find_value_on_line():
    instructions = ['123 -> x', '456 -> y', 'x AND y -> d', 'x OR y -> e', 'x LSHIFT 2 -> f', 'y RSHIFT 2 -> g',
                    'NOT x -> h', 'NOT y -> i']
    wire_dictionary = part_1.create_dictionary(instructions)
    assert part_1.find_value_on_line(wire_dictionary, 'x') == 123
    assert part_1.find_value_on_line(wire_dictionary, 'y') == 456
    assert part_1.find_value_on_line(wire_dictionary, 'd') == 72
    assert part_1.find_value_on_line(wire_dictionary, 'e') == 507
    assert part_1.find_value_on_line(wire_dictionary, 'f') == 492
    assert part_1.find_value_on_line(wire_dictionary, 'g') == 114
    assert part_1.find_value_on_line(wire_dictionary, 'h') == 65412
    assert part_1.find_value_on_line(wire_dictionary, 'i') == 65079

File: 2015/Day_07/Ruby/test_part_1.rb
require_relative "part_1"
require 'test/unit'

class TestWireKit < Test::Unit::TestCase
    
    def test_create_dictionary
        instructions = ['123 -> x', '456 -> y', 'x AND y -> d', 'x OR y -> e', 'x LSHIFT 2 -> f', 'y RSHIFT 2 -> g', 'NOT x -> h', 'NOT y -> i']
        wire_dictionary = create_dictionary(instructions)
        assert_equal("123", wire_dictionary['x'])
        assert_equal("456", wire_dictionary['y'])
        assert_equal("x AND y", wire_dictionary['d'])
    end
    
    def test_break_up_equation
        equation = "x AND y"
        assert_equal(['x', 'AND', 'y'], break_up_equation(equation))
        equation = "x LSHIFT 2"
        assert_equal(['x', 'LSHIFT', '2'], break_up_equation(equation))
        equation = "NOT x"
        assert_equal(['NOT', 'x'], break_up_equation(equation))
        equation = "lx"
        assert_equal(['lx'], break_up_equation(equation))
        equation = "NOT lx"
        assert_equal(['NOT', 'lx'], break_up_equation(equation))
        equation = "2 LSHIFT x"
        assert_equal(['2', 'LSHIFT', 'x'], break_up_equation(equation))
    end
    
    def test_find_value_on_line
        instructions = ['123 -> x', '456 -> y', 'x AND y -> d', 'x OR y -> e', 'x LSHIFT 2 -> f', 'y RSHIFT 2 -> g',
                    'NOT x -> h', 'NOT y -> i']
        @all_wires = create_dictionary(instructions)
        assert_equal(123, find_value_on_line('x'))
        assert_equal(72, find_value_on_line('d'))
        assert_equal(507, find_value_on_line('e'))
        assert_equal(492, find_value_on_line('f'))
    end
end
